### PR TITLE
Feature/swap transfer batch additional fee

### PR DESCRIPTION
**Pull request by pakrentos**
# Changes
- Added ADAR commission to swap_transfer_batch
- Refactored swap_transfer_batch
- Refactored tests
- Changed ratio of reserves in liquidity_source mock (part of the tests' fix)

### CODE REVIEW

File: pallets/liquidity-proxy/src/test_utils.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
+use crate::mock::{adar, AccountId, Assets, DEXId, LiquidityProxy};
```
Comment (wer1st) [inline pallets/liquidity-proxy/src/test_utils.rs:1]: Add copyright header

File: pallets/liquidity-proxy/src/lib.rs [hunk 2]
```diff
@@ -1545,6 +1549,168 @@ impl<T: Config> Pallet<T> {
...
+            fail!(Error::<T>::ForbiddenFilter);
+        }
+
+        let (
+            SwapOutcome {
+                amount: executed_input_amount,
+                ..
+            },
+            _,
+        ) = Self::inner_exchange(
```
Comment (wer1st) [inline pallets/liquidity-proxy/src/lib.rs:1587]: Suggest you to get my changes from #327 for this extrinsic

File: pallets/liquidity-proxy/src/lib.rs [hunk 3]
```diff
@@ -1545,6 +1549,185 @@ impl<T: Config> Pallet<T> {
...
+    fn transfer_batch_tokens_unchecked(
+        sender: &T::AccountId,
+        output_asset_id: &T::AssetId,
+        receivers: Vec<BatchReceiverInfo<T::AccountId>>,
+        remainder_per_receiver: Balance,
+    ) -> Result<(), DispatchError> {
+        let mut unique_batch_receivers: BTreeSet<T::AccountId> = BTreeSet::new();
+        fallible_iterator::convert(receivers.into_iter().map(|val| Ok(val))).for_each(|receiver| {
+            // extrinsic fails if there are duplicate account ids in the same output asset id
+            if !unique_batch_receivers.insert(receiver.account_id.clone()) {
```
Comment (vovac12) [inline pallets/liquidity-proxy/src/lib.rs:1637]: We had discussion with Dmitriy about that and decided to remove this check

File: pallets/liquidity-proxy/src/lib.rs [hunk 4]
```diff
@@ -1545,6 +1549,185 @@ impl<T: Config> Pallet<T> {
...
+                            &selected_source_types,
+                            dex_id,
+                            &filter_mode,
+                            out_amount,
+                        )?
+                    } else {
+                        (out_amount, 0)
+                    };
+
+                let adar_commssion_portion =
```
Comment (vovac12) [inline pallets/liquidity-proxy/src/lib.rs:1707]: Why not calculate sum of input amounts and call `calculate_adar_commission` on this number?

File: pallets/liquidity-proxy/src/lib.rs [hunk 5]
```diff
@@ -1769,104 +1953,29 @@ pub mod pallet {
...
-                    max_input_amount = max_input_amount
-                        .checked_sub(executed_input_amount)
-                        .ok_or(Error::<T>::SlippageNotTolerated)?;
+            assets::Pallet::<T>::transfer_from(
+                &input_asset_id,
+                &who,
+                &T::GetADARAccountId::get(),
+                adar_commission,
+            )
+            .map_err(|_| Error::<T>::InvalidReceiversInfo)?;
```
Comment (vovac12) [inline pallets/liquidity-proxy/src/lib.rs:1977]: Maybe it's better to use SlippageNotTolerated or some additional error type?

**Review by vovac12 — APPROVED** [review]

**Pull request merged**